File: palpng/resolution.js



function PangResolution (app, json) {
	this.app = app;
	this.json = json;
	this.name = this.json.name;
	this.height = this.json.height;
	this.width = this.json.width;
}



PangResolution.prototype.getName = function () {
	return this.name;
}

PangResolution.prototype.getNameX = function () {
	return `${this.width}x${this.height}`;
}

PangResolution.prototype.getHeight = function () {
	return this.height;
}

PangResolution.prototype.getWidth = function () {
	return this.width;
}



module.exports = PangResolution;

File: cpp/resolution.cpp
#include "pang/resolution.hpp"

#include "pang/pangapp.hpp"



pang::Resolution::Resolution (App *app, nj::json json)
{
	this->app = app;
	this->name = json["name"];
	this->height = json["height"];
	this->width = json["width"];
}

pang::Resolution::~Resolution ()
{
	//
}



int pang::Resolution::getHeight ()
{
	return this->height;
}

int pang::Resolution::getWidth ()
{
	return this->width;
}

std::string pang::Resolution::getName ()
{
	return this->name;
}

std::string pang::Resolution::getSuffix ()
{
	return this->name;
}


